pint-pkg/src/build.rs

```diff
@@ -41,7 +41,7 @@ pub struct BuildOptions {
     /// contract unique.
     ///
     /// If "salt" is provided for a library package, an error is emitted.
-    pub salt: [u8; 32],
+    pub salts: HashMap<manifest::ManifestFile, [u8; 32]>,
```

Comment:
It might be enough to make this `manifest::Manifest` instead of using `ManifestFile` (`ManifestFile` derefs to `Manifest` btw).

I'm still not 100% using `Manifest` as a key like this is the best approach, but we need something that is reliably unique, so I guess it will do for now? In the build plan we use `pint_pkg::plan::PinnedId` for this, but it's quite specific to the planning step.

Reply:
I'll open an issue to track this :) Thanks!